### ISSUE TITLE
Question: Angular 4 + Observables

### ISSUE THREAD

**Issue by BorntraegerMarc** [gilf/angular2-indexeddb#17]
Hi there
Short question: Is this lib ready for production use (with AoT) and is Angular 4 supported?
Thx

**Comment by gilf** [comment]
Since it is only a service it should work in Angular 4. AoT is already supported in the last version.

**Comment by BorntraegerMarc** [comment]
@gilf Thanks for the quick answer! Are Observable supported in the lib? Do not find explicit documentation about that.

My goal is to use the library as offline support: As soon as another component in my app sets new values in the db this library should notify that a value changed and send a event.
Would that be possible?

**Comment by gilf** [comment]
@BorntraegerMarc there is no need to use Observable. One of the choices I made when created the library was to use promises instead on observables. The main reason is that all requests to indexeddb are only single so you don't need an object that expects streams to do the work. 

Regarding your question, when updating a value in db you can wire to the then function anything you want so it won't be hard to send a notification on the promise success callback.

**Comment by BorntraegerMarc** [comment]
@gilf Yep, saw the promise implementations. That's great!

But for clarifications: My main goal was to implement offline capabilities. And there observers would have come in quite handy. So my ideal process would look something like this:

1. Open up app and make backend REST Call
2. (async) Store REST call response in indexed DB
3. Reload website on bad connection or no connection at all
4.1. (async) load stored response from DB
4.2. (async) Store REST call response in indexed DB
5. Update client as soon as the new data has been loaded and stored in the DB

I think for this scenario a observable pattern would be good, correct?

So to summarize: If I want to implement a behavior like described, then I would need call `db.getAll` from this library every 100ms and if something changes I would notify the observers. Would that work?

Do you see any performance problems?

**Comment by gilf** [comment]
Hi,

I suggest to start with the onLine attribute in the navigator object and the online/offline events - https://developer.mozilla.org/en/docs/Online_and_offline_events 

This will help you to identify offline scenarios (not accurate but sufficient).

**Comment by BorntraegerMarc** [comment]
Thanks again for the answer @gilf. I know the events but unfortunately they are indeed not sufficient for our use case. That is why I need to update the values "on the fly".

Which is achievable with a library like https://www.webcomponents.org/element/PolymerElements/app-storage/app-indexeddb-mirror for example. But unfortunately Google Polymer does not play nicely along with Angular.

So theoretically I would be able to implement the Observable pattern with this library. But do you see any performance issues arising?
And would you be open for a PR?

**Comment by gilf** [comment]
Theoretically yes :) 
I'm open for PRs. Any support is welcome.

**Comment by Cl3r1k** [comment]
Can you provide an example with add method and Observable?
Trying to do it so: 
calling the addItem method in component
`this._indexedDbService.addItem('Item desc').subscribe((newItem) => {
            console.log('newItem is: ', newItem);
        });`

and the implementation in service: 
`addItem(itemName: string): Observable<string> {

        return Observable.of(this.db.add(this.storeName, { title: itemName, complete: false }).then((newItem) => {
            console.log('addItem - added new item: ', newItem);
            return newItem;
        }, (error) => {
            return Observable.throw(error);
        })
        );
    }`

but answer is: 
`newItem is:  ZoneAwarePromise {__zone_symbol__state: null, __zone_symbol__value: Array(0)}
            __zone_symbol__state: true
            __zone_symbol__value: {key: 24, value: {…}}`

How to pass back only 'newItem' as Observable<string>?
